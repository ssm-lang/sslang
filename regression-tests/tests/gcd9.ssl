gcd a b c cout =
  while deref a != deref b
  
    after 1, cout <- deref a + 48
    wait cout
    after 1, cout <- 32
    wait cout
    after 1, cout <- deref b + 48
    wait cout
    after 1, cout <- 10
    wait cout
  
    if deref a < deref b
      after 1, cout <- 60 // <
      wait cout
      b <- deref b - deref a
    else
      after 1, cout <- 62 // >
      wait cout
      a <- deref a - deref b

  c <- deref a

main cin cout =
  let c = new 0
  gcd (new 4) (new 18) c cout
  after 1, cout <- deref c + 48
  wait cout
  after 1, cout <- 10
  wait cout
  