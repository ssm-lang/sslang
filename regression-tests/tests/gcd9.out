4 B
<4 >
<4 :
<4 6
<4 2
>2
